cou/apps/app.py

```diff
@@ -335,19 +418,17 @@ def apt_source_codename(self) -> Optional[OpenStackRelease]:
     def channel_codename(self) -> OpenStackRelease:
         """Identify the OpenStack release set in the charm channel.
 
+        When application comes from charm store, it's not possible to determine the
+        OpenStack release codename and in that case it will be considered as ussuri.
         :return: OpenStackRelease object
         :rtype: OpenStackRelease
         """
-        try:
-            # get the OpenStack release from the channel track of the application.
-            os_track_release_channel = OpenStackRelease(self.channel.split("/", maxsplit=1)[0])
-        except ValueError:
-            logger.debug(
-                "The current channel of '%s' does not exist or is unexpectedly formatted",
-                self.name,
-            )
-            os_track_release_channel = self.current_os_release
-        return os_track_release_channel
+        self.channel = self.status.charm_channel
```

Comment:
Why we are configuring `channel` here? I think it should be in `post_init`. What do you say?

Reply:
Problem is that for versionless applications we end up calling `channel_codename` without setting the channel before. My next commit tries to have better comment of this issue. See what you think about it.